contracts/Chaosnet.sol

```diff
@@ -0,0 +1,78 @@
+pragma solidity 0.8.9;
+
+/// @title Chaosnet
+/// @notice This is a beta staker program for stakers willing to go the extra
+/// mile with monitoring, share their logs with the dev team, and allow to more
+/// carefully monitor the bootstrapping network. As the network matures, the
+/// beta program will be ended.
+contract Chaosnet {
+  /// @notice Indicates if the chaosnet is active. The chaosnet is active
+  /// after the contract deployment and can be ended with a call to
+  /// `deactivateChaosnet()`. Once deactivated chaosnet can not be activated
+  /// again.
+  bool public isChaosnetActive;
+
+  /// @notice Indicates if the given operator is a beta operator for chaosnet.
+  mapping(address => bool) public isBetaOperator;
+
+  /// @notice Address controlling chaosnet status and beta operator addresses.
+  address public chaosnetMaestro;
+
+  event BetaOperatorsAdded(address[] operators);
+
+  event ChaosnetMaestroRoleTransferred(
+    address oldChaosnetMaestro,
+    address newChaosnetMaestro
+  );
+
+  event ChaosnetDeactivated();
+
+  constructor() {
+    _transferChaosnetMaestro(msg.sender);
```

Comment:
why not
```
chaosnetMaestro = msg.sender
```
?
Calling a function that assignes and reassignes vars seems like too much in the constructor that is invoked only once anyway. Besides `chaosnetMaestro` is public, so we can always check the address in case you wanted to emit the event. 
This way we can merge `transferChaosnetMaestroRole` & `_transferChaosnetMaestro` into one function.

Reply:
I think the advantage of using the current approach is that the current address for this role can be checked only by looking at the events which could be an advantage for tools like The Graph. Here is an example from OpenZeppelin [`Ownable`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/7a14f6c5953a1f2228280e6eb1dfee8e5c28d79a/contracts/access/Ownable.sol#L28-L30), they are doing it exactly the same.

Reply:
I would avoid code duplication. I know we were not consistent and we were usually not emitting events when setting initial values in the constructor but in retrospect, I think it's worth it. All changes of the value, including the initial one, can then be tracked by just looking at events with no additional initialization "if"s.